pep-9999.rst

```diff
@@ -175,8 +190,33 @@ performed):
                 pep508 += " @ " + vcs + "+" requirement[vcs]
                 if "revision" in requirement:
                     pep508 += "@" + revision
+        markers = ""
+        for variable in (
+            "os_name",
+            "sys_platform",
+            "platform_machine",
+            "platform_python_implementation",
+            "platform_release",
+            "platform_system",
+            "platform_version",
+            "python_version",
+            "python_full_version",
+            "implementation_name",
+            "implementation_version",
+            "extra",
+        ):
+            key = variable.replace("_", "-")
+            if key in requirement:
+                if markers:
+                    markers += " and "
+                markers += variable + " " + requirement[key]
         if "markers" in requirement:
-            pep508 += "; " + requirement["markers"]
+            markers_explicit = requirement["markers"]
+            if markers:
+                markers_explicit = " and (" + markers_explicit + ")"
+            markers += markers_explicit
+        if markers:
+            pep508 += "; " + markers
```

Comment:
The `;` needs to be prefixed with a whitespace.

Reply:
For consistency with the rest of the PEP 508 conversion, yeah probably. It doesn't need it however, as AFAIK all white-space in a PEP-508 is optional (for example my [`seddy` `setup.cfg`](https://github.com/EpicWink/seddy/blob/master/setup.cfg#L26))

Reply:
The issue is URLs, ie direct references. This will break since `;` is valid in a URL. Thus, whitespace is used as a terminal character.

Reply:
Yeah I just realised that